geometry: make arbitrary_point of Plane run through all points

normalhuman opened the pull request:
Changes Plane.arbitrary_point so that it is able to represent any point in the plane, by way of linear equations with two parameters `u` and `v` (these appear to be traditional names of surface parameters, as opposed to t for curve parameter). Closes #13799.

The method `random_point` is changed accordingly. Incidentally, it was found that a plane could previously be created with zero normal vector. A check and a test are added to prevent this.

#### Backwards compatibility is preserved

Originally, `arbitrary_point` had one parameter and traced a circle in the plane. This behavior is retained if `arbitrary_point` gets 0 or 1 arguments, for the sake of compatibility. An improvement in this case is that the parametric formula for the circle is much cleaner now.

normalhuman committed:
geometry: make arbitrary_point of Plane run through all points

Changes Plane.arbitrary_point so that it is able to represent
any point in the plane, by way of linear equations with two parameters
u and v (these appear to be traditional names of surface parameters, as
opposed to t for curve parameter). Closes #13799.

The method random_point is changed accordingly. Incidentally, it was
found that a plane could previously be created with zero normal vector.
A check and a test are added to prevent this.

normalhuman committed:
Corrected errors in newly added tests in test_plane

Turned out the tests were skipped as "slow" when I ran bin/test

ylemkimon reviewed (approved):
+1, though deprecation process may be necessary.

smichr commented:
> deprecation process may be necessary

(Thinking out loud...) We should be free to change the arbitrary point routine as long as it continues to return an arbitrary point. The current routine returns *more* arbitrary points than before as it now covers (potentially) any point in the plane. The change is that it now takes 2 parameters. Since an arbitrary parameter was chosen before for t we should still be able to do so. What's different is that the arbitrary points were on a circle before. Is there any way to let t still give a circle -- in a simpler manner -- and another variable, say 'r', give the radius? Or is the non-x-y-plane circle just too complicated to work with? Could you just replace u with `r*cos(2*pi*t)` and v with `r*sin(2*pi*t)` in your formulation? (Maybe not, but is there something like that that could be done?)

normalhuman commented:
I would expect a linear object to have linear parameterization. Giving that up because of compatibility does not appeal to me. I suggest a compromise: the method should work as before if it gets 0 or 1 parameter, and changes to new behaviour when given 2 parameters.

normalhuman committed:
Retain backward compatibility in plane.arbitrary_point

If given 1 or 0 parameters, this method still returns a unit circle
(although much quicker and with a much cleaner formula). Only if given
two parameters does it change to linear parameterization of the plane.
Thus, the behavior of current version is still supported.

normalhuman commented:
Based on the above discussion, I changed the method so that it traces a circle when given 0 or 1 arguments, and the entire plane if given 2 arguments. Thus the previous behavior of the method is not broken anymore. 

I hope this eliminates the need for deprecation process.

normalhuman committed:
Correct errors in new tests (only hit with --slow testing)

smichr commented:
Nice job! I think deprecation is unnecessary.

smichr commented:
Anyone wanting to make the point on the circle cover the entire plane can right-multiply the point by a radius:

```
>>> Point(1,2,3)*r
Point3D(r, 2*r, 3*r)
```

normalhuman commented:
Almost. One would have to subtract p1, multiply by r, and add p1 back.
